UI/Selenium/Steps/SelectHearingSteps.cs

```diff
@@ -29,9 +29,13 @@ public class SelectHearingSteps : ObjectFactory
         [When(@"I click on copy hearing id to clipboard")]
         public void WhenIClickOnCopyHearingIdToClipboard()
         {
-            ExtensionMethods.MoveToElement(Driver, SelectHearingPage.Quicklinks, _scenarioContext);
-            ExtensionMethods.FindElementWithWait(Driver, SelectHearingPage.Hearingidtoclipboard, _scenarioContext).Click();
-            ApplicationData.hearingID = new TextCopy.Clipboard().GetText();
+            if (ExtensionMethods.WaitForPageLoad(Driver, SelectHearingPage.Quicklinks, _scenarioContext))
+            {
+                ExtensionMethods.MoveToElement(Driver, SelectHearingPage.Quicklinks, _scenarioContext);
+                ExtensionMethods.FindElementWithWait(Driver, SelectHearingPage.Hearingidtoclipboard, _scenarioContext).Click();
+                ApplicationData.hearingID = new TextCopy.Clipboard().GetText();
```

Comment:
we're using ScenarioContext to share data between steps. I think we should use same approach here instead of ApplicationData

Reply:
Changed